examples/createContract/src/spec.md

```diff
@@ -19,11 +21,18 @@ storage
 if
 
     #newAddr(ACCT_ID, Nonce_Mom) == KID
+    Id == keccakIntList(                                                                                                 \
+        keccak(#parseByteStackRaw("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)")) \
+        keccak(#parseByteStackRaw("Kid V1"))                                                                             \
+        keccak(#parseByteStackRaw("1"))                                                                                  \
+        VChainId                                                                                                         \
+        KID                                                                                                              \
+    )
+    VCallDepth < 1024
 
 iff
 
     VCallValue == 0
-    VCallDepth < 1024
```

Comment:
why is the constraint on `VCallDepth` not required for this spec, but still needed for the others?

Reply:
The `VCallDepth` constraint needs to be in an `if` block for the assembly creates. Solidity must provide some more convenience when using `new`. I couldn't say exactly though - more research required.

Reply:
🤔

Reply:
Is this confirmed and/or are we satisfied with the current state of research? :)

I personally have no idea and am happy to defer to yours.

Reply:
Solidity bubbles up the reverts if done with new, in assembly you would need to check the return value pushed on the stack and revert manually